### PR TITLE
simplify kubectl testing factory

### PR DESCRIPTION
**Pull request by deads2k**
This simplifies and strips down the fake factory and collapses us down to a single fake!

@kubernetes/sig-cli-maintainers @juanvallejo 
/assign @pwittrock 
/assign @soltysh 

```release-note
NONE
```

### CODE REVIEW

**Comment by deads2k** [comment]
Now with a single fake factory!

File: pkg/kubectl/cmd/testing/fake.go [hunk 1]
```diff
@@ -235,221 +232,77 @@ func (d *fakeCachedDiscoveryClient) ServerResources() ([]*metav1.APIResourceList
 }
 
 type TestFactory struct {
```
Comment (deads2k) [inline pkg/kubectl/cmd/testing/fake.go]: The meat is in this file. Everything else is ripples

**Review by juanvallejo — APPROVED** [review]

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**